### PR TITLE
doc: add contributing guidelines

### PR DESCRIPTION
**Pull request by davidruvolo51**
What are the main changes you did:
- Add guidelines for contributing to the EMX2 repository

how to test:
- explain here what to do to test this (or point to unit tests)

todo:
- [x] updated docs in case of new feature
- [ ] add link in the frontend development guide

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+### Code contributions
+
+Anyone is welcome to contribute to the EMX2 code base. We've outlined this process below.
+
+#### Pre-development
+
+We have a few guides that provide information on how EMX2 is structured, the coding guidelines, and other practical information. These will help you get started with writing code and to ensure it is consistent with the rest of the code base.
+
+- [Development guidelines](https://molgenis.github.io/molgenis-emx2/#/molgenis/dev_guidelines)
```
Comment (mswertz) [inline CONTRIBUTING.md:31]: bit dangerous to have so many lists. Ideally we would simply refer to 'developer manual' right?
Then the links don't go stale.

And then ensure developer manual has nice starting page to guide people to right parts. E.g. 'develop apps'.

Comment (davidruvolo51) [reply CONTRIBUTING.md]: This is a good point. I added dev landing page and created a link in the contributing guidelines.

**Review by mswertz — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+We have created several [developer guides](https://molgenis.github.io/molgenis-emx2/#/dev) that provide information on how EMX2 is structured, the coding guidelines, and other practical information. These will help you get started with writing code and to ensure it is consistent with the rest of the code base.
+
+The easiest way to get started is to start with the latest version of EMX2. Clone the main repository and create a new branch.
+
+```bash
+git clone https://github.com/molgenis/molgenis-emx2
+
+# create a new branch
+git switch -c feat/<name-of-feature>
```
Comment (chinook25) [inline CONTRIBUTING.md:37]: Is kebab what we want to use for branches? I tend to use camel casing. I have also seen other use snake. What do we prefer?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+- All tests pass and all errors/warnings are fixed
+- Add documentation on how to use the new feature or any breaking change. Documentation is located in the `docs/`. If you are unsure where to add the docs, please let us know in the PR.
+- Commit all changes and give it a clear and concise message
+- Push to the molgenis-emx2 repository and open a new PR. In the PR, provide as much information about the contribution as possible. This includes an overview of your contribution and what was changed/added. Please indicate if there any breaking changes.
+
+## The submission process
+
+All contributions (bug report, feature request, pull request) will be reviewed by a member of the MOLGENIS-EMX2 team. Then, the following steps will take place.
+
+1. **Triage of the contribution**: We will first triage the request/bug report/PR. It will be reviewed to make sure it is clear. W may request additional information.
```
Comment (chinook25) [inline CONTRIBUTING.md:59]: W -> missing letter

File: docs/molgenis/README.md [hunk 4]
```diff
@@ -1,20 +1,23 @@
...
 
 ## Differences with previous version of MOLGENIS
 
-Core differences with molgenis/molgenis include that it is simpler to setup and operate, it allows multiple data
-schemas, it has more sophisticated data modelling features, and for developers it presents data using a self-documenting
-graphql api. Obviously it doesn't have all features of molgenis/molgenis yet, so that is being maintained as well until
-adequate migration paths exist.
+[molgenis-emx2](http://github.com/molgenis/molgenis-emx2) (or EMX2) is the newest iteration of the MOLGENIS platform. It builds on the previous version ([molgenis/molgenis](https://github.com/molgenis/molgenis), or EMX1) by:
+
+- **Simpler setup and operation**: We simplified the core structure and EMX2 is built using PostgreSQL, Java, GraphQL, and Vue.
```
Comment (chinook25) [inline docs/molgenis/README.md:19]: Do we need to mention gradle here? Maybe `is built by gradle using...`?

File: docs/molgenis/README.md [hunk 5]
```diff
@@ -1,20 +1,23 @@
...
+- **Enhanced data modelling features**: we enabled the option for multiple data schemas, schema templates, and FAIR data improvements (e.g., the `ONTOLOGY` data type). For developers, the data API specification is now auto-generated. Any changes to the schema will be automatically documented in the GraphQL API
+- **Improved data APIs**: In EMX2, we've incorporated [GraphQL](https://graphql.org). This allows users to write more advanced queries
 
-Use the menu on the left to navigate through this document.
+We are still in the process of adding EMX1 features into EMX2. For now, EMX1 will be maintained but not actively developed.
```
Comment (chinook25) [inline docs/molgenis/README.md:23]: Maybe it is good to also add that we plan to deprecate it at some point in the future?

File: docs/molgenis/dev_apps.md [hunk 6]
```diff
@@ -4,7 +4,9 @@ One of the features of MOLGENIS EMX2, is the ability to create your own frontend
 
 This guide will provide information on how to create a new app or how to contribute to an existing application. You will also find information on the tools that are used in MOLGENIS EMX2 and how to configure your app to deploy it in EMX2. In the last section (the appendices), you will find a step-by-step guides for developing apps.
 
-Before you begin, we encourage developers to read our [Development Guidelines](./dev_guidelines) to ensure consistency of all interfaces. If you notice any issue with this guide (e.g, broken example, outdated information, etc.), please open an issue on at [molgenis-emx2/issues](https://github.com/molgenis/molgenis-emx2/issues).
+Before you begin, we encourage developers to read our [Development Guidelines](./dev_guidelines) to ensure consistency of all interfaces.
+
+If you get stuck, have a look at the [Troubleshooting guide](./dev_troubleshooting). If you notice any issue with this guide (e.g, broken example, outdated information, etc.), please open an issue on at [molgenis-emx2/issues](https://github.com/molgenis/molgenis-emx2/issues).
```
Comment (chinook25) [inline docs/molgenis/dev_apps.md:9]: Do we need to link to the contributing here for the guidelines of opening issues (might be overkill here)?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+The easiest way to get started is to start with the latest version of EMX2. Clone the main repository and create a new branch.
+
+```bash
+git clone https://github.com/molgenis/molgenis-emx2
+
+# create a new branch
+git switch -c feat/<name-of-feature>
+```
+
+Regarding branch names, we typically use these patterns for naming branches: `feat/*` for a new feature or improvement, and `fix/` for bugs. Keep the branch names short and concise.
```
Comment (chinook25) [inline CONTRIBUTING.md:40]: Do we need to mention `chore/*` ?

File: docs/molgenis/dev_apps.md [hunk 8]
```diff
@@ -60,7 +62,7 @@ git clone https://github.com/molgenis/molgenis-emx2
 git switch -c feat/my-new-app
 ```
 
-Regarding branch names, we typically use these patterns for naming branches: `feat/*` for a new feature or improvement, and `fix/` for bugs.
+Regarding branch names, we typically use these patterns for naming branches: `feat/*` for a new feature or improvement, and `fix/` for bugs. For further information, please see the [Basic concepts](./dev_basics) guide.
```
Comment (chinook25) [inline docs/molgenis/dev_apps.md:65]: chore?

**Review by chinook25 — APPROVED** [review]
Just some minor questions and 1 typo, but overall, good work :D

File: CONTRIBUTING.md (hunk 3 above)
Comment (davidruvolo51) [reply CONTRIBUTING.md:59]: Good catch

File: docs/molgenis/README.md (hunk 5 above)
Comment (davidruvolo51) [reply docs/molgenis/README.md:23]: This is good point. I added more information on this

File: docs/molgenis/dev_apps.md (hunk 6 above)
Comment (davidruvolo51) [reply docs/molgenis/dev_apps.md:9]: Good idea. I added a link to the contributing guide

File: CONTRIBUTING.md (hunk 7 above)
Comment (davidruvolo51) [reply CONTRIBUTING.md:40]: Yes, this should also be in here. Instead, I put a link to to the Basics Guide as it's defined there

File: CONTRIBUTING.md (hunk 2 above)
Comment (davidruvolo51) [reply CONTRIBUTING.md:37]: I don't have a preference for this. The most important aspect is the branch prefix (e.g., feat, fix, etc.).

Comment (chinook25) [reply CONTRIBUTING.md:37]: Me neither, so let's leave it like this.

**Pull request merged**